### DISCUSSION TITLE
Best models to train on DreamBooth? (Community Help)

### DISCUSSION THREAD

**Discussion by meunumerotim333** [TheLastBen/fast-stable-diffusion#1763] [Q&A]
Hey guys, since I've discovered we can only train models on DreamBooth Colab that have diffusers..

**What's the best models you guys recommend to use?**

This Model here Called "DreamShaper" I've read on CIVITAI that has diffusers.. but give me an error.. so I would like the help of community to use models without keep been frustrated trying

![image](https://user-images.githubusercontent.com/121369033/224852175-91045ee3-7dee-4d95-9f84-3e1bf48bbd68.png)

**Comment by gilroff** [comment]
im curious too. i tried urpm and got errors.

**Reply by Tanguyu2** [reply]
Uber13 to train on it, good results, full version
https://drive.google.com/file/d/14-mzGYWYyQDtiYhneGxD-9KDoke6_8p_/view?usp=sharing

**Comment by LyceumCorp** [comment]
I found a solution by d8ahazard that can help to continue training of such models. Haven't tried it yet though.
https://github.com/d8ahazard/sd_dreambooth_extension/discussions/794

**Reply by gilroff** [reply]
Oh thanks, ill try it out.

**Reply by meunumerotim333** [reply]
Thank you, I'll also try :)

**Reply by meunumerotim333** [reply]
> I found a solution by d8ahazard that can help to continue training of such models. Haven't tried it yet though. [d8ahazard/sd_dreambooth_extension#794](https://github.com/d8ahazard/sd_dreambooth_extension/discussions/794)

I'm getting Cuda error on Colab. Did you marked some option that the guy didn't mention on his text tutorial? 

What I'm doing wrong?

![image](https://user-images.githubusercontent.com/121369033/225178465-c2a41668-2c49-4757-b2bc-1333a2959532.png)

**Reply by meunumerotim333** [reply]
Not even selecting "Save as float16" is working.. getting Cuda error

**Reply by LyceumCorp** [reply]
> > I found a solution by d8ahazard that can help to continue training of such models. Haven't tried it yet though. [d8ahazard/sd_dreambooth_extension#794](https://github.com/d8ahazard/sd_dreambooth_extension/discussions/794)
> 
> I'm getting Cuda error on Colab. Did you marked some option that the guy didn't mention on his text tutorial?
> 
> What I'm doing wrong?

As I mentioned I haven't tried it yet. But I got the same error today while trying to train with LORA. Not sure 100% Need investigation. Try asking on yt.

**Reply by meunumerotim333** [reply]
Its working using the light version on SD 1.5. Thanks. But also .safetensors version of the file works well on DreamBooth and I didn't know

**Reply by Tanguyu2** [reply]
You solve it with colab premium GPU, _moni moni_

**Reply by Mossbraker** [reply]
Sorry, I already posted this below in sort of the wrong spot. For the issue of running out of memory when merging checkpoints, you can add the --lowvram argument in the "Start Stable-Diffusion" section of the colab. So, change the code from:
```python
    !python /content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py $share --api --disable-safe-unpickle --enable-insecure-extension-access --no-download-sd-model --no-half-vae  --ckpt "$model" --xformers $auth --disable-console-progressbars
  else:
    !python /content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py $share --api --disable-safe-unpickle --enable-insecure-extension-access --no-download-sd-model --no-half-vae  --ckpt-dir "$model" --xformers $auth --disable-console-progressbars
except:
   !python /content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py $share --api --disable-safe-unpickle --enable-insecure-extension-access --no-download-sd-model --no-half-vae --xformers $auth --disable-console-progressbars`
```
to:
```python
    !python /content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py $share --api --disable-safe-unpickle --enable-insecure-extension-access --no-download-sd-model --no-half-vae  --ckpt "$model" --xformers $auth --disable-console-progressbars --lowvram
  else:
    !python /content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py $share --api --disable-safe-unpickle --enable-insecure-extension-access --no-download-sd-model --no-half-vae  --ckpt-dir "$model" --xformers $auth --disable-console-progressbars --lowvram
except:
   !python /content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py $share --api --disable-safe-unpickle --enable-insecure-extension-access --no-download-sd-model --no-half-vae --xformers $auth --disable-console-progressbars --lowvram
```
and then you shouldn't have the CUDA out of memory issue anymore. And you don't need colab premium. Remove the --lowvram arguments when you aren't merging checkpoints.

**Comment by Mossbraker** [comment]
If you need the diffusers version of a model, check huggingface - most popular models have been uploaded there, plus plenty of others. Here is DreamShaper: https://huggingface.co/Lykon/DreamShaper/tree/main
On that page, click "</> Use in Diffusers" and you'll find the model link that you should be able to use on this colab.

For the issue of running out of memory when merging checkpoints, you can add the --lowvram argument in the "Start Stable-Diffusion" section of the colab. So, change the code from:
```python
    !python /content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py $share --api --disable-safe-unpickle --enable-insecure-extension-access --no-download-sd-model --no-half-vae  --ckpt "$model" --xformers $auth --disable-console-progressbars
  else:
    !python /content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py $share --api --disable-safe-unpickle --enable-insecure-extension-access --no-download-sd-model --no-half-vae  --ckpt-dir "$model" --xformers $auth --disable-console-progressbars
except:
   !python /content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py $share --api --disable-safe-unpickle --enable-insecure-extension-access --no-download-sd-model --no-half-vae --xformers $auth --disable-console-progressbars`
```
to:
```python
    !python /content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py $share --api --disable-safe-unpickle --enable-insecure-extension-access --no-download-sd-model --no-half-vae  --ckpt "$model" --xformers $auth --disable-console-progressbars --lowvram
  else:
    !python /content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py $share --api --disable-safe-unpickle --enable-insecure-extension-access --no-download-sd-model --no-half-vae  --ckpt-dir "$model" --xformers $auth --disable-console-progressbars --lowvram
except:
   !python /content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py $share --api --disable-safe-unpickle --enable-insecure-extension-access --no-download-sd-model --no-half-vae --xformers $auth --disable-console-progressbars --lowvram
```
and then you shouldn't have the CUDA out of memory issue anymore. Remove the --lowvram arguments when you aren't merging checkpoints.

**Comment by meunumerotim333** [comment]
Guys, thank you.

I've solved doing this, a friend have teach me how to train models that gives error. I don't know the reason why it works.

![image](https://user-images.githubusercontent.com/121369033/225997701-2999fecb-7fc2-4a54-a34c-285bdc285f97.png)

![image](https://user-images.githubusercontent.com/121369033/225997839-a27622cf-e05b-4e1a-801e-ef503e9a1f6d.png)

Doing this, and not using a specific file, it works magically. Not only for this model, but a lot of others that was giving the same error.

**Reply by Mossbraker** [reply]
This is a slightly simpler way of doing what I suggested, i.e.:
> If you need the diffusers version of a model, check huggingface - most popular models have been uploaded there, plus plenty of others. Here is DreamShaper: https://huggingface.co/Lykon/DreamShaper/tree/main
On that page, click "</> Use in Diffusers" and you'll find the model link that you should be able to use on this colab.

and you probably explained more clearly as well. And as I said, this works because the path you specify (i.e. Lykon/DreamShaper) is a directory containing the diffusers.

**Reply by meunumerotim333** [reply]
Sorry if I skiped your tip, I really didn't know how to do it. It was my first time doing, didn't know this was possible and my english isn't fluent to notice instantly, so I've skiped, sorry and thank you.
